### DISCUSSION TITLE
307 temporary redirect error - Wordpress import

### DISCUSSION THREAD

**Discussion by philam** [umputun/remark42#1733] [Q&A]
Hi there,

I've successfully set up Remark42 on an Oracle Cloud setup.

When I try to run the Wordpress import script described in the docs [here](https://remark42.com/docs/backup/migration/) I get the following error via SSH:

```
docker exec -it remark42 import -p wordpress -f /srv/var/Wordpress.xml -s remark42
remark42 v1.12.1-1f2500f-20230821T11:59:46

[INFO]  {cmd/import.go:26 cmd.(*ImportCommand).Execute} import /srv/var/Wordpress.xml (wordpress), site remark42

[ERROR] {app/main.go:52 main.main.func1} failed with error response "307 Temporary Redirect",
error response "307 Temporary Redirect"
```

Any ideas on what might be causing this?

**Comment by paskal** [comment]
According to what I see in the code, the problem is likely with the auth. Are you sure you set the `ADMIN_PASSWORD` on the Remark42 server and restarted it afterwards?
I'll try replicating the no password scenario and add a meaningful error message to ensure it's not as vague as `failed with error response "307 Temporary Redirect"`.

**Reply by paskal** [reply]
I found and fixed some problems with import and backup in #1734. However, I couldn't reproduce the problem you're seeing. Could you please also post the logs of Remark42 itself when you run the import?

~Also, I believe you don't need to supply the `-p` flag, as running inside Docker will take the password from the environment.~  never mind, I saw that `-p` is a provider and not a password.

**Reply by paskal** [reply]
The most likely reason for the problem is the wrong `REMARK_URL` directive, so the request is made for the wrong URL. Then fixing that variable or providing the flag `--url=http://remark42.example.org` to the import command would fix the situation.

**Reply by paskal** [reply]
I found the root cause. Your `REMARK_URL` is `http://` while it should be `https://`. I'll improve the error logging for that case.

**Reply by paskal** [reply]
But I am not entirely sure. 307 redirects, and the current code doesn't work with redirects. Still, it is different than what you saw: instead of displaying the status code 307, it follows the redirect but does so without the auth headers, so it receives 401 Unauthorized errors after that, even with valid credentials.

Once you figure out the problem, could you please try to reproduce it and write the conditions under which it happened?
